report/onionscanreport.go

```diff
@@ -65,8 +65,20 @@ type OnionScanReport struct {
 	SMTPFingerprint string `json:"smtpFingerprint"`
 	SMTPBanner      string `json:"smtpBanner"`
 
+	ProtocolInfoList []ProtocolInfo `json::"protocolInfoList"`
+
 	NextAction string `json:"lastAction"`
-	TimedOut   bool
+	TimedOut   bool   `json:"timedOut"`
+}
+
+type ProtocolInfo struct {
+	Type string      `json:"type"`
+	Port uint        `json:"port:`
+	Info interface{} `json:"info"`
+}
+
+func (osr *OnionScanReport) AddProtocolInfo(protocolType string, protocolPort uint, protocolInfo interface{}) {
+	osr.ProtocolInfoList = append(osr.ProtocolInfoList, ProtocolInfo{protocolType, protocolPort, protocolInfo})
```

Comment:
I'm having a try at porting over some other scans to this system and just noticed that `osr.ProtocolInfoList` is not initialized to an empty list, so it will be null instead of [] if nothing is found. Dunno if that's a problem.